kitsune/wiki/tests/test_views.py

```diff
@@ -684,3 +685,144 @@ def test_it_works(self):
+        assert trans_content in doc_content
+
+    def test_header_locales_not_in_translated_list(self):
+        """Test while the document is not translated into requested locale nor the document
+        translated into any of the locale of ACCEPT_LANGUAGE header. But we have a fallback_locale
+        Mentioned in kitsune/settings.py for one of the locale of ACCEPT_LANGUAGE header"""
+
+        # A Client with de, an, ja language choices in ACCEPT_LANGUAGE Header
+        header = 'de,an;q=0.7,ja;q=0.3,'
+        # Create a translated document in es locale version.
+        # And try to get data of de locale vesion of document with above mentioned client
+        doc_content = self.get_data_from_translated_document(header=header,
+                                                             create_doc_locale='es',
+                                                             req_doc_locale='de')
+        # As the document is not translated into de locale version, also there are no locale
+        # in client header which the document is translated. But we have a fallback locale
+        # mentioned in kitsune/settings.py for the an locale. it is es. As the document is
+        # translated into es version, the user should see es version of the document
+        en_content = 'This artcile is in English'
+        trans_content = 'This artcile is translated into es'
+        assert en_content not in doc_content
+        assert trans_content in doc_content
+
+    def test_custom_locale_mapping(self):
+        """Test while the document is not translated into requested locale but
+        We have custom wiki fallback for the requested locale.
+        The Custom wiki fallback locale mentioned in kitsune/wiki.config.py"""
+
+        # Create a translated document in bn-BD locale and try to get bn-IN version
+        # of the document. While we the client have no language choices.
+        doc_content = self.get_data_from_translated_document(header=None,
+                                                             create_doc_locale='bn-BD',
+                                                             req_doc_locale='bn-IN')
+        # As the Document is not translated into bn-IN and there is no locale of
+        # ACCEPT_LANUGAE header which the document to, But there is a custom wiki fallback
+        # mapping available for bn-IN. the fallback locale is bn-BD. So the user should
+        # see the document in bn-BD.
+        en_content = 'This artcile is in English'
+        trans_content = 'This artcile is translated into bn-BD'
+        assert en_content not in doc_content
+        assert trans_content in doc_content
+
+    def test_custom_locale_mapping_for_header_locale(self):
+        """Test while the document is not translated into any of the locale of ACCEPT_LANGUAGE
+        Header. But we have fallback locale mentioned in Custom wiki fallback locale for one
+        of the locale of ACCEPT_LANGUAGE Header"""
```

Comment:
Doc strings should have a short summary in one line. If there needs to be more information, it should go after a blank line for example:

```python
"""
Test that configured locale fallbacks are used.

With a document not translated into any of the locales available in
ACCEPT_LANGUAGE, test that locales listed as fallbacks in `wiki/config.py` are
used.
"""
```